MaxText/layers/moe.py

```diff
@@ -910,11 +907,11 @@ def dense_matmul(self, inputs, gate_logits, pre_bias_logits, w0_kernel, w1_kerne
       with jax.named_scope("w_sum"):
         if is_llama4_decoder_layer:
           weights = self.reshape_and_update_weights(jnp.ones_like(top_k_weights), top_k_indices)
-        output = jnp.einsum(
-            "BSEM,BSE -> BSM",
-            intermediate_layer,
-            weights,
-        ).astype(self.dtype)
+          output = jnp.einsum(
```

Comment:
Extra space for the `output` that causes one test fails. Could you move it out of `if` statement?

Reply:
That's a weird one; ok I manually adjusted spacing but it needs to be in the `if` statement otherwise `weights` isn't guaranteed to be set